packages/core/crates/core-types/src/protocol.rs

```diff
@@ -24,6 +27,15 @@ pub type Tag = u16;
 /// Represent a default application tag if none is specified in `send_packet`.
 pub const DEFAULT_APPLICATION_TAG: Tag = 0;
 
+/// Trait for linking and resolving the corresponding `OffchainPublicKey` and on-chain `Address`.
+#[async_trait(? Send)] // TODO: the resolver should not be async once detached from the DB ? Also make it `Send` once DB is `Send` too
+pub trait PeerAddressResolver {
+    /// Tries to resolve off-chain public key given the on-chain address
+    async fn resolve_packet_key(&self, onchain_key: &Address) -> Option<OffchainPublicKey>;
+    /// Tries to resolve on-chain public key given the off-chain public key
+    async fn resolve_chain_key(&self, offchain_key: &OffchainPublicKey) -> Option<Address>;
+}
+
```

Comment:
Ideally implemented as a separate DB.

Reply:
Agreed. And should have an LRU cache in memory